src/resources/x509.cc

```diff
@@ -27,7 +27,24 @@
 #if !defined(TOIT_FREERTOS) || CONFIG_TOIT_CRYPTO
 namespace toit {
 
-Object* X509ResourceGroup::parse(Process* process, const uint8_t* encoded, size_t encoded_size) {
+// A simple whitespace detector tuned for PEM format that doesn't accept exotic
+// whitespace characters.
+static inline bool is_white_space(int c) {
+  return c == ' ' or c == '\n' or c == '\r';
+}
+
+bool X509ResourceGroup::is_pem_format(const uint8* data, size_t length) {
+  const char HEADER[] = "-----BEGIN ";
+  while (length > 0 && is_white_space(data[0])) {
+    length--;
+    data++;
+  }
+  if (length < sizeof(HEADER)) return false;
```

Comment:
Pick sizeof(HEADER) or strlen(HEADER) -- maybe put it in a local const -- and use that consistently.